src/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.5)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+find_package(PkgConfig REQUIRED)
+
+set(GSTREAMER_MINIMUM_VERSION 1.8)
+pkg_check_modules(GST1_TEST gstreamer-1.0)
+
+if (NOT GST1_TEST_FOUND )
+    message( FATAL_ERROR "GStreamer 1.8 required." )
+endif()
+if ( ${GST1_TEST_VERSION} VERSION_LESS ${GSTREAMER_MINIMUM_VERSION} )
+    message( FATAL_ERROR "GStreamer 1.8 required. Found ${GST1_TEST_VERSION} instead." )
+endif()
+
+pkg_check_modules(GST REQUIRED
+    gstreamer-rtsp-1.0
+    gstreamer-1.0
+    gstreamer-rtsp-server-1.0
+    gstreamer-app-1.0
+)
+
+project(dcm)
+
+add_definitions(
+    -std=c++11
+    -Wall
+    -Wextra
+    -Wno-unused-parameter
+    -Wno-unused-variable
+    -Wno-missing-field-initializers
+)
+
+file(GLOB SRC_FILES *.cpp)
```

Comment:
As far as I know this is generally discouraged. We should list the files manually.